curve-amm/src/tests.rs

```diff
@@ -0,0 +1,84 @@
+use codec::Encode;
+use fungible_token_messages::*;
+use gstd::{vec, vec::Vec, ActorId, String};
+use gtest::{Program, RunResult, System};
+const USERS: &'static [u64] = &[3, 4, 5];
+
+fn init(sys: &System) {
+    sys.init_logger();
+
+    let usdc_token = Program::from_file(
+        &sys,
+        "../../apps/target/wasm32-unknown-unknown/release/fungible_token.wasm",
+    );
+    let usdt_token = Program::from_file(
+        &sys,
+        "../../apps/target/wasm32-unknown-unknown/release/fungible_token.wasm",
+    );
+    let lp_token = Program::from_file(
+        &sys,
+        "../../apps/target/wasm32-unknown-unknown/release/fungible_token.wasm",
+    );
+    let curve_amm = Program::from_file(
+        &sys,
+        "../../apps/target/wasm32-unknown-unknown/release/curve_amm.wasm",
+    );
+
+    let res = usdc_token.send(
+        USERS[0],
+        InitConfig {
+            name: String::from("USDC Stable Coin"),
+            symbol: String::from("USDC"),
+        },
+    );
+    assert!(res.log().is_empty());
+    let res = usdt_token.send(
+        USERS[0],
+        InitConfig {
+            name: String::from("USDT Stable Coin"),
+            symbol: String::from("USDT"),
+        },
+    );
+    assert!(res.log().is_empty());
+    let res = lp_token.send(
+        USERS[0],
+        InitConfig {
+            name: String::from("LP Token"),
+            symbol: String::from("LPT"),
+        },
+    );
+    assert!(res.log().is_empty());
+
+    let usdc_program_id = usdc_token.id();
+    let usdt_program_id = usdt_token.id();
+    let lp_program_id = lp_token.id();
+    let mut token_accounts: Vec<u8> = vec![];
+    token_accounts.extend_from_slice(&usdc_program_id.as_slice());
+    token_accounts.extend_from_slice(&usdt_program_id.as_slice());
+    token_accounts.extend_from_slice(&lp_program_id.as_slice());
+
+    let res = curve_amm.send(
+        USERS[0],
+        crate::CurveAmmInitConfig {
+            token_accounts,
+            amplification_coefficient: 10000,
+            fee: 0,
+            admin_fee: 0,
+        },
+    );
+
+    assert!(res.log().is_empty());
+}
+
+fn add_admin(ft: Program, admin: &ActorId, owner: u64) -> RunResult {
+    return ft.send(owner, fungible_token_messages::Action::AddAdmin(*admin));
+}
+
+#[test]
+fn add_admin_test() {
```

Comment:
That test is for fungible token contract that has already been covered with tests, I think there is no need to test it here :)

Reply:
yes I am aware , I was just trying things and need help from dev team so pushed this code.